### PR TITLE
docs: Add documentation for GeM pooling

### PR DESCRIPTION
**Pull request by LMMilliken**
This pr adds the section on GeM pooling that was previously removed

---

- [ ] This PR references an open issue
- [x] I have added a line about this change to CHANGELOG

### CODE REVIEW

File: docs/advanced-topics/advanced-losses-optimizers-and-poolers.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 (advanced-losses-optimizers-poolers)=
-# {octicon}`mortar-board` Advanced losses and optimizers
+# {octicon}`mortar-board` Advanced losses, optimizers and poolers
```
Comment (scott-martens) [inline docs/advanced-topics/advanced-losses-optimizers-and-poolers.md:2]:
```suggestion
# {octicon}`mortar-board` Advanced loss functions, optimizers and poolers
```
Using "losses" as a pure noun that way is a little jarring to me.

File: docs/advanced-topics/advanced-losses-optimizers-and-poolers.md [hunk 2]
```diff
@@ -78,12 +78,30 @@ Each color represents a different class. You can see how all of the loss functio
...
-In cases where each class contains very few examples of each class, but many classes, `ArcFaceLoss` and `CosFaceLoss` struggle to separate every class,
-and doesn't perform as well as `TripletMarginLss`.  
-Another case where `ArcFaceLoss` and `CosFaceLoss` struggle is when attempting to embed documents of a class they were not trained on. 
-Since these loss functions operate by creating centroids, and then attempting to 'classify' input documents by embedding them in a space close to a centroid, 
-data that does not belong to a class with an existing centroid may erroneously be given an embedding similar to that of a class it does not belong to.  
-In cases such as these, `TripletMarginLoss` may result in a greater improvement in performance thatn `ArcFaceLoss` or `CosFaceLoss`.
\ No newline at end of file
+Pooling layers are layers in a machine learning model that are used to reduce the dimensionality of data. This is usually done for one of two reasons: to remove unnecessary information contained within an embedding of a larger size, or when a model outputs multiple embeddings and only one embedding is needed. Typically this is done in two ways, average pooling or max pooling.
```
Comment (scott-martens) [inline docs/advanced-topics/advanced-losses-optimizers-and-poolers.md:83]:
```suggestion
Pooling layers are layers in a machine learning model that are used to reduce the dimensionality of data. This is usually done for one of two reasons: to remove unnecessary information contained within an embedding of a larger size, or when a model outputs multiple embeddings and only one embedding is needed. Typically there are  two ways. to do this: average pooling or max pooling.
```

File: docs/advanced-topics/advanced-losses-optimizers-and-poolers.md [hunk 3]
```diff
@@ -78,12 +78,30 @@ Each color represents a different class. You can see how all of the loss functio
...
+While a model may have many pooling layers within it, it is unwise to replace a pooling layer with another unless it is the last layer of the model.
+
+### GeM pooling
+
+`GeM` (Generalised Mean) pooling is an advanced pooling technique that has found popularity in computer vision and face recognition tasks.
```
Comment (scott-martens) [inline docs/advanced-topics/advanced-losses-optimizers-and-poolers.md:88]:
```suggestion
`GeM` (Generalised Mean) pooling is an advanced pooling technique that is popular for computer vision and face recognition tasks.
```

File: docs/advanced-topics/advanced-losses-optimizers-and-poolers.md [hunk 4]
```diff
@@ -78,12 +78,30 @@ Each color represents a different class. You can see how all of the loss functio
...
+In cases where your chosen model does have a pooling layer as its last layer, Finetuner allows you to replace the default pooler with a `GeM` pooling layer.
+Currently, all of our `text-to-text` and `image-to-image` models support replacing the pooling layer.
+For a list of all models that fit these categroies, see the [Backbone Model](../walkthrough/choose-backbone.md) section.
```
Comment (scott-martens) [inline docs/advanced-topics/advanced-losses-optimizers-and-poolers.md:91]:
```suggestion
For a list of all models that support replacing the pooling layer, see the [Backbone Model](../walkthrough/choose-backbone.md) section.  
```

File: docs/advanced-topics/advanced-losses-optimizers-and-poolers.md [hunk 5]
```diff
@@ -78,12 +78,30 @@ Each color represents a different class. You can see how all of the loss functio
...
+While a model may have many pooling layers within it, it is unwise to replace a pooling layer with another unless it is the last layer of the model.
+
+### GeM pooling
+
+`GeM` (Generalised Mean) pooling is an advanced pooling technique that has found popularity in computer vision and face recognition tasks.
+In cases where your chosen model does have a pooling layer as its last layer, Finetuner allows you to replace the default pooler with a `GeM` pooling layer.
+Currently, all of our `text-to-text` and `image-to-image` models support replacing the pooling layer.
+For a list of all models that fit these categroies, see the [Backbone Model](../walkthrough/choose-backbone.md) section.  
+
+The `GeM` pooler has two adjustable parameters, a scaling parameter `p` and an epsilon `eps`.
```
Comment (scott-martens) [inline docs/advanced-topics/advanced-losses-optimizers-and-poolers.md:93]:
```suggestion
The `GeM` pooler has two adjustable parameters: a scaling parameter `p` and an epsilon `eps`.
```

File: docs/advanced-topics/advanced-losses-optimizers-and-poolers.md [hunk 6]
```diff
@@ -78,12 +78,30 @@ Each color represents a different class. You can see how all of the loss functio
...
+At `p = 1`, the `GeM` pooler will act like an average pooler.
+As `p` increases, more weight is given to larger values, making it act more like max pooling.
+`eps` is used to clamp values to be slightly above 0, altering this won't result in much change to the performance.
```
Comment (scott-martens) [inline docs/advanced-topics/advanced-losses-optimizers-and-poolers.md:96]:
```suggestion
`eps` is used to clamp values to be slightly above 0, and altering this won't result in much change to the performance.
```

File: docs/advanced-topics/advanced-losses-optimizers-and-poolers.md [hunk 7]
```diff
@@ -78,12 +78,30 @@ Each color represents a different class. You can see how all of the loss functio
...
+By default, `p=3` and `eps=1e-6`, you can specify the pooler and adjust these parameters in a dictionary provided to the `model_options` parameter:
```
Comment (scott-martens) [inline docs/advanced-topics/advanced-losses-optimizers-and-poolers.md:97]:
```suggestion
By default, `p=3` and `eps=1e-6`. You can specify the pooler and adjust these parameters in a dictionary provided to the `model_options` parameter:
```

**Comment by bwanglzu** [comment]
if Scott approves, then feel free to merge

**Comment by bwanglzu** [comment]
let's add a small "insight" to tell users, combine `GeM` pooling together with `ArcFaceLoss` delivers better results.

**Review by scott-martens — APPROVED** [review]
LGTM

**Pull request merged**